### PR TITLE
Some edits to chapters 2,3,4

### PR DESCRIPTION
**Pull request by rjleveque**
Some suggested changes in the first three chapters after the preface.

I just noticed the Preface comes out as Chapter 1 in the pdf, maybe we need to fix this.

### CODE REVIEW

**Comment by ketch** [comment]
>I just noticed the Preface comes out as Chapter 1 in the pdf, maybe we need to fix this.

Yes; I didn't find a nice way to automate that so I think it's best to just do it manually.

File: exact_solvers/acoustics_demos.py [hunk 1]
```diff
@@ -40,30 +40,38 @@ def decompose_q(p,u,K,rho):
...
     axes[0].legend(['$r_1$','$r_2$'],loc=3)
     axes[0].plot([0,0],[-2,2],'--k',alpha=0.2)
     axes[0].plot([-2,2],[0,0],'--k',alpha=0.2)
+    axes[0].set_xlabel('p')
+    axes[0].set_ylabel('u')
 
     
-    axes[1].plot([0,p],[0,u],'k',lw=3)    
+    axes[1].plot([0,p],[0,u],'k-',lw=3,markersize=8)
+    #axes[1].plot([p],[u],'ko',markersize=8)
```
Comment (ketch) [inline exact_solvers/acoustics_demos.py]: I'm guessing you meant to eventually delete this line and the other one also commented out just below.

Comment (rjleveque) [reply exact_solvers/acoustics_demos.py]: Yes, thanks.  Now done.

**Comment by rjleveque** [comment]
This is ready to merge if it looks ok to you.  I'll do separate PRs for changes to other chapters.

**Pull request merged**